### PR TITLE
util: ignore some sql in statement summary

### PR DESCRIPTION
**Pull request by djshow832**
### What problem does this PR solve?

Problem Summary:

Ignore some sql in statement summary tables: begin, use, explain, set, and other.
Because these statements won't cause performance problems, but they will make some noises.

### What is changed and how it works?

What's Changed:
Ignore some sql in statement summary tables: begin, use, explain, set, and other.

How it Works:

### Related changes

- PR to update `pingcap/docs`/`pingcap/docs-cn`:
- Need to cherry-pick to the release branch

### Check List 

Tests 

- Unit test
- Manual test (add detailed scripts or steps below)
```
mysql> use test;
Database changed
mysql> explain select 1;
+-------------------+---------+------+---------------+---------------+
| id                | estRows | task | access object | operator info |
+-------------------+---------+------+---------------+---------------+
| Projection_3      | 1.00    | root |               | 1->Column#1   |
| └─TableDual_4     | 1.00    | root |               | rows:1        |
+-------------------+---------+------+---------------+---------------+
2 rows in set (0.00 sec)

mysql> explain analyze select 1;
+-------------------+---------+---------+------+---------------+-----------------------------------------+---------------+---------+------+
| id                | estRows | actRows | task | access object | execution info                          | operator info | memory  | disk |
+-------------------+---------+---------+------+---------------+-----------------------------------------+---------------+---------+------+
| Projection_3      | 1.00    | 1       | root |               | time:3.158µs, loops:2, Concurrency:OFF  | 1->Column#1   | 0 Bytes | N/A  |
| └─TableDual_4     | 1.00    | 1       | root |               | time:690ns, loops:2                     | rows:1        | N/A     | N/A  |
+-------------------+---------+---------+------+---------------+-----------------------------------------+---------------+---------+------+
2 rows in set (0.00 sec)

mysql> select stmt_type, digest_text from performance_schema.events_statements_summary_by_digest\G
*************************** 1. row ***************************
  stmt_type: explain
digest_text: explain analyze select ?
*************************** 2. row ***************************
  stmt_type: select
digest_text: select database ( )
*************************** 3. row ***************************
  stmt_type: show
digest_text: show tables
*************************** 4. row ***************************
  stmt_type: show
digest_text: show databases
*************************** 5. row ***************************
  stmt_type: select
digest_text: select variable_value from mysql . tidb where variable_name = ?
5 rows in set (0.00 sec)
```

Side effects

N/A

### Release note 

- Ignore some statements that won't cause performance problems in statement summary.

### CODE REVIEW

File: util/stmtsummary/statement_summary.go [hunk 1]
```diff
@@ -884,3 +885,19 @@ func convertEmptyToNil(str string) interface{} {
 	}
 	return str
 }
+
+// NeedRecord judges whether this type of statement should be summarized.
+func NeedRecord(stmtNode ast.StmtNode) bool {
+	switch stmtNode.(type) {
+	case ast.DMLNode, ast.DDLNode:
+		return true
+	case *ast.BeginStmt, *ast.DeallocateStmt, *ast.FlushStmt, *ast.SetConfigStmt, *ast.SetDefaultRoleStmt,
```
Comment (AilinKid) [inline util/stmtsummary/statement_summary.go:894]: commit stmt?

File: util/stmtsummary/statement_summary.go [hunk 2]
```diff
@@ -884,3 +885,19 @@ func convertEmptyToNil(str string) interface{} {
 	}
 	return str
 }
+
+// NeedRecord judges whether this type of statement should be summarized.
+func NeedRecord(stmtNode ast.StmtNode) bool {
```
Comment (AilinKid) [inline util/stmtsummary/statement_summary.go:890]: cause it return bool, how about rename it to `CheckNeedRecord`

File: util/stmtsummary/statement_summary.go (hunk 1 above)
Comment (djshow832) [reply util/stmtsummary/statement_summary.go:894]: `commit` is important, because in optimistic transactions, `commit` is always slow.

**Comment by wwar** [comment]
I don't think this change is a good idea. At a minimum it is not MySQL compatible behavior (these statements are in statement summary).

Users rely on this summary table to understand a complete profile of server activity. I would want to know if my applications are sending noisy statements to the server, because even if they are very low cost in TiDB, the network protocol is synchronous. It could indicate that I need to tune a driver to be less noisy (JDBC for example is notoriously bad by default). The latency will add up for applications.

**Comment by AilinKid** [comment]
> I don't think this change is a good idea. At a minimum it is not MySQL compatible behavior (these statements are in statement summary).
> 
> Users rely on this summary table to understand a complete profile of server activity. I would want to know if my applications are sending noisy statements to the server, because even if they are very low cost in TiDB, the network protocol is synchronous. It could indicate that I need to tune a driver to be less noisy (JDBC for example is notoriously bad by default). The latency will add up for applications.
> 
> **Edit:** The "mysql" way to ignore these is sys schema (sys.statement_analysis is ordered by total execution time).

PTAL @djshow832

**Comment by djshow832** [comment]
I discussed with the person who suggested it, and we decided not to change it. So I'll close it now.

**Pull request closed without merging**